hoot-core/src/main/cpp/hoot/core/io/OsmApiChangeset.cpp

```diff
@@ -866,7 +866,48 @@ bool XmlChangeset::calculateChangeset(ChangesetInfoPtr& changeset)
   return changeset->size() > 0;
 }
 
-ChangesetInfoPtr XmlChangeset::splitChangeset(ChangesetInfoPtr changeset)
+bool XmlChangeset::matchesPlaceholderFailure(const QString& hint,
+                                             long& member_id, ElementType::Type& member_type,
+                                             long& element_id, ElementType::Type& element_type)
+{
+  //  Placeholder node not found for reference -145213 in way -5687
+  //  Placeholder Way not found for reference -12257 in relation -51
+  QRegExp regPlaceholder("Placeholder (node|way|relation) not found for reference (-?[0-9]+) in (node|way|relation) (-?[0-9]+)", Qt::CaseInsensitive);
+  if (hint.contains(regPlaceholder) && regPlaceholder.captureCount() == 4)
+  {
+    //  Get the node/way/relation type and id that caused the failure
+    member_type = ElementType::fromString(regPlaceholder.cap(1));
+    bool success = false;
+    member_id = regPlaceholder.cap(2).toLong(&success);
+    if (!success)
+      return success;
+    //  Get the node/way/relation type and id that failed
+    element_type = ElementType::fromString(regPlaceholder.cap(3));
+    element_id = regPlaceholder.cap(4).toLong(&success);
+    return success;
+  }
+  return false;
+}
```

Comment:
Functions to parse element type and ID from the error messages that I've seen from the OSM API that I'm able to deal with immediately.